vetomint/src/progress.rs

```diff
@@ -82,64 +74,105 @@ pub(super) fn progress(
                 round,
                 ..
             } => {
-                let total_voting_power = height_info.validators.iter().sum::<VotingPower>();
                 if round != state.round {
                     return None;
                 }
-                let voting_power = height_info.validators[signer as usize];
-                state.votes.insert(round, {
-                    let mut votes = state
-                        .votes
-                        .get(&round)
-                        .unwrap_or(&Default::default())
-                        .clone();
-                    votes.prevotes_total += voting_power;
-                    votes.prevotes_favor.insert(
-                        proposal,
-                        votes.prevotes_favor.get(&proposal).unwrap_or(&0) + voting_power,
-                    );
-                    votes
-                });
-                if state.votes[&round].prevotes_total * 6 > total_voting_power * 5
-                    && state.step == ConsensusStep::Prevote
-                {
-                    on_5f_prevote(height_info, state, round)
-                } else if state.votes[&round].prevotes_total * 3 > total_voting_power * 2
-                    && (state.step == ConsensusStep::Prevote
-                        || state.step == ConsensusStep::Precommit)
-                {
-                    on_4f_prevote(height_info, state, round)
-                } else {
-                    Vec::new()
+                let voted_information = state.prevote_history[&round].get(&signer);
```

Comment:
voted_information -> vote_information

Reply:
fixed line 80, and same issues in progress.rs